src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -1549,13 +1549,15 @@ export class CustomTreeViewDragAndDrop implements ITreeDragAndDrop<ITreeItem> {
 		this.dndController = controller;
 	}
 
-	private handleDragAndLog(dndController: ITreeViewDragAndDropController, itemHandles: string[], uuid: string, dragCancellationToken: CancellationToken): Promise<VSDataTransfer | undefined> {
+	private handleDragAndLog(dndController: ITreeViewDragAndDropController, itemHandles: string[], uuid: string, dataTransfer: DataTransfer, dragCancellationToken: CancellationToken): Promise<VSDataTransfer | undefined> {
 		return dndController.handleDrag(itemHandles, uuid, dragCancellationToken).then(additionalDataTransfer => {
 			if (additionalDataTransfer) {
 				const unlistedTypes: string[] = [];
-				for (const item of additionalDataTransfer.entries()) {
-					if ((item[0] !== this.treeMimeType) && (dndController.dragMimeTypes.findIndex(value => value === item[0]) < 0)) {
-						unlistedTypes.push(item[0]);
+				for (const [mimeType, entry] of additionalDataTransfer.entries()) {
+					if (mimeType !== this.treeMimeType && !dndController.dragMimeTypes.some(value => value === mimeType)) {
+						unlistedTypes.push(mimeType);
+					} else {
+						dataTransfer.setData(mimeType, entry.value);
```

Comment:
I am surprised this works! I though that the original `DataTransfer` has been disposed at this point since we are in async code. For regular editors, we use this:

https://github.com/mjbvz/vscode/blob/28f9cc3d4c65dc0d3f8da6cf903d255cd4c82f5c/src/vs/workbench/browser/parts/views/treeView.ts#L1579-L1580

and then do this to retrieve the value on drop:

https://github.com/mjbvz/vscode/blob/28f9cc3d4c65dc0d3f8da6cf903d255cd4c82f5c/src/vs/workbench/browser/parts/editor/editorDropTarget.ts#L341-L359

Do we need to do something similar for notebooks?

Reply:
Good point. I think I missed that the value returned in my testing was not actually the value being set by the extension. Will push a new commit with a different approach